### PR TITLE
Add golangci lint

### PR DESCRIPTION
**Pull request by elezar**
This PR adds an initial golangci lint configuration and also addresses some minor lint errors.

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    - gocritic
+    - godot
+    - gofmt
+    - goimports
+    - misspell
+    - stylecheck
+
+run:
+  tests: true
+  timeout: 1m
```
Comment (ArangoGutierrez) [inline .golangci.yaml:17]: is 1 min enough? maybe 5 or 10, github action instances are slow some times

Comment (ArangoGutierrez) [reply .golangci.yaml:17]: oh I see everything is disabled, then 1 min is ok for now

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

File: .golangci.yaml (hunk 1 above)
Comment (elezar) [reply .golangci.yaml]: Note. There are still some linters that are enabled by default: `errcheck` and `gosimple` for example.

These are the ones that trigger the other fixes required in this changset.

I have bumped the timeout to 10m though.

**Review by ArangoGutierrez — APPROVED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (ArangoGutierrez) [reply .golangci.yaml]: DO you think, the lint check should be an extra github action?

**Pull request merged**